containers/kratos/entrypoint.sh

```
@@ -0,0 +1,44 @@
+# Create a working directory that the ory user can write to
+WORK_DIR="/tmp/kratos-config"
+mkdir -p "$WORK_DIR"
+
+# Copy the config file to the working directory
+cp /etc/config/kratos/kratos.yml "$WORK_DIR/kratos.yml"
+
+# Check if template exists and we have environment variables set
+if [ -f "/etc/config/kratos/kratos.yml.template" ] && [ -n "$KRATOS_PUBLIC_URL" ]; then
+    echo "Using templated configuration..."
+    # Export all KRATOS_, DISCORD_, and STEAM_ variables for envsubst
+    export $(env | grep -E '^(KRATOS_|DISCORD_|STEAM_)' | cut -d= -f1)
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)